services/notify/src/api.js

```diff
@@ -89,3 +91,194 @@ builder.declare({
   await this.notifier.irc(input);
   res.sendStatus(200);
 });
+
+builder.declare({
+  method: 'post',
+  route: '/blacklist/add',
+  name: 'addBlacklistAddress',
+  // scopes: {
+  //   if: 'channelRequest',
+  //   then: 'notify:irc-channel:<channel>',
+  //   else: 'notify:irc-user:<user>',
+  // },
+  input: 'notification-address.yml',
+  title: 'Blacklist Given Address',
+  description: [
+    'Add the given address to the notification blacklist. The address',
+    'can be of either of the three supported address type namely pulse, email',
+    'or IRC(user or channel). Addresses in the blacklist will be ignored',
+    'by the notification service.',
+  ].join('\n'),
+}, async function(req, res) {
+  // The address to blacklist
+  let address = Object.assign({}, {
+    notificationType: req.body.notificationType,
+    notificationAddress: req.body.notificationAddress,
+  });
+
+  try {
+    await this.BlacklistedNotification.create(address);
+  } catch (e) {
+    if (e.name === 'EntityAlreadyExistsError') {
+      return res.reportError(
+        'EntityAlreadyExists',
+        'Notification address already exists in the blacklist',
+        {}
+      );
+    } else {
+      throw e;
+    }
+  }
+  res.reply({});
+});
+
+builder.declare({
+  method: 'get',
+  route: '/blacklist/get',
+  name: 'getBlacklistAddress',
+  input: 'notification-address.yml',
+  //output: 'notification-address.yml',
+  //scopes: 'secrets:get:<name>',
+  title: 'Read Blacklisted Notification',
+  description: [
+    'Read the notification, associated with some address, from the blacklist.',
```

Comment:
Seeing how we will have to check the blacklist for existence of an address time and again won't it be more efficient to use this method rather than `list()`?